logic/overworld.py

```diff
@@ -65,50 +65,50 @@ def __init__(self):
+        Location().add(GoldLeaf(0x2C5)).connect(castle_inside, AND(BOMB, attack_hookshot_powder))  # in the castle, bomb wall to show enemy
+        Location().add(GoldLeaf(0x2C6)).connect(castle_inside, OR(BOOMERANG, AND(POWER_BRACELET, attack_hookshot)))  # in the castle, spinning spikeball enemy
 
-        animal_town = Location().connect(center_area, OR(FLIPPERS, PEGASUS_BOOTS))
+        animal_town = Location().connect(center_area, OR(FLIPPERS, HOOKSHOT, AND(PEGASUS_BOOTS, OR(BOMB, BOOMERANG, MAGIC_POWDER, MAGIC_ROD, SWORD)))) # passage under river blocked by bush
         Location().add(Seashell(0x0DA)).connect(animal_town, SHOVEL)  # owl statue at the water
-        desert = Location().add(AnglerKey()).connect(animal_town, bush)  # Note: We removed the walrus blocking the desert.
-        Location().add(HeartPiece(0x2E6)).connect(desert, AND(BOMB, HOOKSHOT))  # cave in the upper right of animal town
+        desert = Location().add(AnglerKey()).connect(animal_town, bush)  # Note: We removed the walrus blocking the desert. TODO: angler key requirements?
+        Location().add(HeartPiece(0x2E6)).connect(desert, AND(BOMB, FEATHER, HOOKSHOT))  # cave in the upper right of animal town
         Location().add(HeartPiece(0x1E8)).connect(desert, BOMB)  # above the quicksand cave
         Location().add(Seashell(0x0FF)).connect(desert, POWER_BRACELET)
 
         # Area below the windfish egg
         below_mountains = Location().connect(graveyard, POWER_BRACELET)
-        into_to_mountains = Location().add(Chest(0x018)).connect(below_mountains, AND(POWER_BRACELET, SWORD))
-        Location().add(Chest(0x2BB)).connect(into_to_mountains, HOOKSHOT)
-        right_mountains_1 = Location().add(Chest(0x28A)).connect(into_to_mountains, PEGASUS_BOOTS)
+        into_to_mountains = Location().add(Chest(0x018)).connect(below_mountains, AND(POWER_BRACELET, SWORD)) # chest outside obstacle cave
+        Location().add(Chest(0x2BB)).connect(into_to_mountains, HOOKSHOT) # chest at obstacles
+        right_mountains_1 = Location().add(Chest(0x28A)).connect(into_to_mountains, PEGASUS_BOOTS) # chest in passage to papahl
         Location().add(HeartPiece(0x1F2)).connect(below_mountains, FLIPPERS)  # cave next to level 4
 
         face_shrine = Location().add(Chest(0x2FC)).connect(animal_town, AND(bush, POWER_BRACELET))
-        Location().add(FaceKey()).connect(face_shrine, OR(BOW, SWORD))
+        Location().add(FaceKey()).connect(face_shrine, OR(BOW, MAGIC_ROD, SWORD))
 
         dungeon6_entrance = Location().connect(animal_town, AND(FLIPPERS, HOOKSHOT))
 
         # Raft game.
         raft_game = Location().add(Chest(0x05C), Chest(0x05D))
-        raft_game.connect(below_mountains, HOOKSHOT)
-        raft_game.connect(center_area, AND(FLIPPERS, HOOKSHOT))
+        raft_game.connect(below_mountains, OR(FLIPPERS, HOOKSHOT)) # flippers from d6 water area to one way cave. Flippers guarantee way back

```

Comment:
FYI, I patched the cave going back to the raft game not to require flippers. And I patched the raft game to be accessible without flippers (normally you cannot jump down the waterfall without flippers)